build.gradle

```diff
@@ -208,16 +208,20 @@ subprojects {
     "org.elasticsearch.plugin:rank-eval-client:${version}": ':modules:rank-eval',
   ]
 
-  for (final Version version : versionCollection.versionsIndexCompatibleWithCurrent) {
-    if (version.branch != null) {
-      final String snapshotProject = ":distribution:bwc-snapshot-${version.branch}"
+  def addBWCSnapshotSubstitutions = { String snapshotProject, Version version ->
+    if (version != null) {
       project(snapshotProject).ext.bwcVersion = version
       ext.projectSubstitutions["org.elasticsearch.distribution.deb:elasticsearch:${version}"] = snapshotProject
       ext.projectSubstitutions["org.elasticsearch.distribution.rpm:elasticsearch:${version}"] = snapshotProject
       ext.projectSubstitutions["org.elasticsearch.distribution.zip:elasticsearch:${version}"] = snapshotProject
     }
   }
 
+  addBWCSnapshotSubstitutions(':distribution:bwc:next-minor-snapshot', versionCollection.nextMinorSnapshot)
+  addBWCSnapshotSubstitutions(':distribution:bwc:staged-minor-snapshot', versionCollection.stagedMinorSnapshot)
+  addBWCSnapshotSubstitutions(':distribution:bwc:next-bugfix-snapshot', versionCollection.nextBugfixSnapshot)
+  addBWCSnapshotSubstitutions(':distribution:bwc:maintenance-bugfix-snapshot', versionCollection.maintenanceBugfixSnapshot)
+
```

Comment:
Ill see what I can do. Im not sure I like passing in the project ext to the VC project, which is used in the addBWCSnapshotSubstitutions method.. but it would be nice to be able to (like the bottom comment about the ifs in the bwc/build.gradle) consolidate the mapping between those in some single place. ill eyeball it, I think it can be better.